crates/api_models/src/errors/types.rs

```diff
@@ -48,21 +106,32 @@ impl ApiErrorResponse {
             | Self::Unprocessable(i)
             | Self::InternalServerError(i)
             | Self::NotImplemented(i)
+            | Self::NotFound(i)
+            | Self::MethodNotAllowed(i)
+            | Self::BadRequest(i)
             | Self::ConnectorError(i, _) => i,
         }
     }
 
     pub(crate) fn error_type(&self) -> &str {
         match self {
-            Self::Unauthorized(_)
-            | Self::ForbiddenCommonResource(_)
-            | Self::ForbiddenPrivateResource(_)
-            | Self::Conflict(_)
-            | Self::Gone(_)
-            | Self::Unprocessable(_)
-            | Self::NotImplemented(_) => "invalid_request",
+            Self::Unauthorized(_) => "invalid_request",
+            Self::ForbiddenCommonResource(_) => "invalid_request",
+            Self::ForbiddenPrivateResource(_) => "invalid_request",
+            Self::Conflict(_) => "invalid_request",
+            Self::Gone(_) => "invalid_request",
+            Self::Unprocessable(_) => "invalid_request",
             Self::InternalServerError(_) => "api",
+            Self::NotImplemented(_) => "invalid_request",
             Self::ConnectorError(_, _) => "connector",
+            Self::MethodNotAllowed(_) => "invalid_request",
+            Self::NotFound(_) => "invalid_request",
+            Self::BadRequest(_) => "invalid_request",
         }
     }
+    // pub fn new() -> Self {
+
+    // }
```

Comment:
Please rebase this branch on top of `main`, it does not contain the latest changes from #524.